src/software/ai/hl/stp/play/ball_placement/ball_placement_play_fsm.h

```diff
@@ -226,24 +230,33 @@ struct BallPlacementPlayFSM
             PickOffWallState_S + Update_E[!wallPickOffDone_G] / pickOffWall_A =
                 PickOffWallState_S,
             PickOffWallState_S + Update_E[wallPickOffDone_G] / startWait_A =
-                ReleaseBallState_S,
+                ReleaseWallPickedState_S,
+
+            ReleaseWallPickedState_S + Update_E[!waitDone_G && !shouldPickOffWall_G] / releaseBall_A =
```

Comment:
we probably don't have time now but some comments here would be nice to explain the flow